flow/scripts/open.tcl

```diff
@@ -80,12 +80,19 @@ if {[env_var_equals GUI_TIMING 1]} {
   }
 
   foreach heatmap {Placement IRDrop Routing RUDY Power} {
-    if {[string equal $heatmap "IRDrop"] || [env_var_exists_and_non_empty PWR_NETS_VOLTAGES]} {
-      catch {
-        gui::set_heatmap $heatmap rebuild 1
-        gui::dump_heatmap $heatmap $::env(REPORTS_DIR)/dummy.png
-      } error
+    if {[string equal $heatmap Routing] && ![grt::have_routes]} {
+      puts "Skipping $heatmap heatmap, no routes available"
+      continue
     }
+    if {[string equal $heatmap IRDrop] && ![env_var_exists_and_non_empty PWR_NETS_VOLTAGES]} {
+      puts "Skipping $heatmap heatmap, no PWR_NETS_VOLTAGES available"
+      continue
```

Comment:
@maliberty How do I skip the IRDrop heatmap when there is no data?

```
 make gui_floorplan
ODB_FILE=./results/nangate45/gcd/base/2_floorplan.odb GUI_SHOW=1 /home/oyvind/OpenROAD-flow-scripts/tools/install/OpenROAD/bin/openroad -gui  -minimize /home/oyvind/OpenROAD-flow-scripts/flow/scripts/open.tcl
QSocketNotifier: Can only be used with threads started with QThread
OpenROAD v2.0-17013-gf7f634f88 
Features included (+) or not (-): +Charts +GPU +GUI +Python
This program is licensed under the BSD-3 license. See the LICENSE file for details.
Components of this program may be licensed under more restrictive licenses which must be honored.
GUI_TIMING=1 reading timing, takes a little while for large designs...
Populating timing paths...OK
Prerendering...
Dumping Placement heatmap...
Dumping IRDrop heatmap...
[ERROR GUI-0072] "IR Drop" is not populated with data.
[ERROR GRT-0010] Instance ctrl.state.out\[0\]$_DFF_P_ is not placed.
[ERROR GUI-0070] Error: open.tcl, 101 GRT-0010
^Cmake: *** [Makefile:1013: gui_2_floorplan.odb] Interrupt
```

Reply:
How do I skip Placement heatmap?

[ERROR GRT-0010] Instance ctrl.state.out\[0\]$_DFF_P_ is not placed.
[ERROR GUI-0070] Error: open.tcl, 105 GRT-0010

Reply:
We don't save the ir drop information in odb so if you are reloading there isn't any.

Reply:
I don't know what that means... I haven't used IR Drop yet. I'm going to remove IRDrop for now if there isn't an easy way to skip it when prerendering should be skipped.

Placement heatmap is more interesting: how do I skip it when there's no placement?

Reply:
You could loop over the instances and see if any are placed.

Reply:
Worked. I haven't tested how well it scales though.